### ISSUE TITLE
Poll Time for HLS Manifest

### ISSUE THREAD

**Issue by dbanbahji** [shaka-project/shaka-player#4771]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
yes

**Is your feature request related to a problem? Please describe.**

yes.  While earlier specs say to use the target duration tag to poll for manifest updates, later specs suggest the last segment duration as the poll time. Sometimes target duration is just a set length 1 or 2 seconds larger than the set segment length. So a manifest that has variable segment length which is very common in ad insertion workflows can sometimes have buffer starvation near the live edge. This should provide more robust support for hls ssai workflows. This is a spec that is followed commonly in commercial players like bitmovin, theoplayer, and OS hlsjs.

**Describe the solution you'd like**
use last downloaded segment duration as the poll time for the next manifest update. A variable poll time.
Basically later official spec versions describe that approach so it is spec approved.

   When a client loads a Playlist file for the first time or reloads a
   Playlist file and finds that it has changed since the last time it
   was loaded, the client MUST wait for at least the duration of the
   last segment in the Playlist before attempting to reload the Playlist
   file again, measured from the last time the client began loading the
   Playlist file.

https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-6.3.4

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Comment by avelad** [comment]
In order to do this we have to change how we internally update playlists. Right now they are all updated at the same time, with this change they should be updated separately because the video and audio segments do not have to last the same time, for example.
